### PR TITLE
fix(utils): fixRequestBody support `multipart/form-data`

### PR DESCRIPTION
**Pull request by JS-mark**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by chimurai** [comment]
Could you add unit tests in:
https://github.com/chimurai/http-proxy-middleware/blob/master/test/unit/fix-request-body.spec.ts

**Comment by JS-mark** [comment]
> Could you add unit tests in: https://github.com/chimurai/http-proxy-middleware/blob/master/test/unit/fix-request-body.spec.ts

ok It has been added.

**Review by JS-mark — COMMENTED** [review]
PTAL

File: src/handlers/fix-request-body.ts [hunk 1]
```diff
@@ -3,6 +3,21 @@ import type * as express from 'express';
 import type { Request } from '../types';
 import * as querystring from 'querystring';
 
+/**
```
Comment (chimurai) [inline src/handlers/fix-request-body.ts:6]: Please move the `handlerFormDataBodyData` to the bottom of the file.

Please remove the `export` of this private function

File: test/unit/fix-request-body.spec.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { ClientRequest } from 'http';
 import * as querystring from 'querystring';
-
-import { fixRequestBody } from '../../src/handlers/fix-request-body';
+import * as FormData from 'form-data';
+import { fixRequestBody, handlerFormDataBodyData } from '../../src/handlers/fix-request-body';
```
Comment (chimurai) [inline test/unit/fix-request-body.spec.ts:4]: Please remove and avoid testing with private `handlerFormDataBodyData` directly

File: test/unit/fix-request-body.spec.ts [hunk 3]
```diff
@@ -52,6 +52,40 @@ describe('fixRequestBody', () => {
...
+
+    jest.spyOn(proxyRequest, 'setHeader');
+    jest.spyOn(proxyRequest, 'write');
+
+    const body = new FormData();
+    body.append('someField', 'some value');
+
+    fixRequestBody(proxyRequest, { body } as Request<express.Request>);
+
+    const expectedBody = handlerFormDataBodyData('multipart/form-data', body);
```
Comment (chimurai) [inline test/unit/fix-request-body.spec.ts:67]: Test will always pass if you use the private `handlerFormDataBodyData` function to generate the expectedBody for assertion.

Please use hardcoded values or jest's inline snapshot to assert the `write` call signature.

File: test/unit/fix-request-body.spec.ts [hunk 4]
```diff
@@ -52,6 +52,40 @@ describe('fixRequestBody', () => {
...
```
Comment (chimurai) [inline test/unit/fix-request-body.spec.ts:84]: Same. Please remove the call to `handlerFormDataBodyData` to generate `expectedBody`

**Review by chimurai — CHANGES_REQUESTED** [review]

File: test/unit/fix-request-body.spec.ts [hunk 5]
```diff
@@ -52,6 +52,40 @@ describe('fixRequestBody', () => {
...
 
+  it('should write when body is not empty and Content-Type is multipart/form-data', () => {
+    const proxyRequest = fakeProxyRequest();
+    proxyRequest.setHeader('content-type', 'multipart/form-data');
+
+    jest.spyOn(proxyRequest, 'setHeader');
+    jest.spyOn(proxyRequest, 'write');
+
+    const body = new FormData();
+    body.append('someField', 'some value');
```
Comment (chimurai) [inline test/unit/fix-request-body.spec.ts:63]: `someField` is added, can you add an assertion?

**Pull request closed without merging**